src/otx/algorithms/common/adapters/mmcv/utils/fp16_utils.py

```diff
@@ -0,0 +1,134 @@
+"""Custom fp16 related modules to enable XPU modules."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import functools
+from inspect import getfullargspec
+from typing import Callable, Iterable, Optional
+
+import torch
+from mmcv.runner.fp16_utils import cast_tensor_type
+from mmcv.utils import IS_NPU_AVAILABLE, TORCH_VERSION, digit_version
+from torch import nn
+
+from otx.algorithms.common.utils import is_xpu_available
+
+try:
+    if is_xpu_available():
+        # import intel_extension_for_pytorch as ipex
+        from torch.xpu.amp import autocast
```

Comment:
Difference between this and mmcv's is only which autocast to use, right?
If then, how about executing code below instead of making new function?
```
if is_xpu_available():
    from torch.xpu.amp import autocast
    from mmcv.runner import fp16_utils
    fp16_utils.autocast = autocast
```